chimney/.jvm/src/test/scala/io/scalaland/chimney/WixSpec.scala

```diff
@@ -308,16 +308,34 @@ object WixSpec extends TestSuite {
           .withFieldConst(_.otherValue, "default")
           .buildTransformer
 
-        EntityDTO(None).transformInto[Entity] ==> Entity(SdlMissingIdPlaceholder, "default")
+        EntityDTO(None).transformInto[Entity] ==> Entity(SdlMissingIdPlaceholderString, "default")
       }
 
-      "apply only to Strings" - {
+      "support UUID placeholder" - {
+        case class Entity(@id(UUIDCompatible, IdGeneration.Auto) id: UUID)
+
+        implicit val t: Transformer[String, UUID] = str => UUID.fromString(str)
+
+        EntityDTO(None).transformInto[Entity] ==> Entity(UUID.fromString(SdlMissingIdPlaceholderUUID))
+        EntityDTO(None).into[Entity].transform ==> Entity(UUID.fromString(SdlMissingIdPlaceholderUUID))
+      }
+
+      "support UUID exception" - {
+        case class Entity(@id(UUIDCompatible, IdGeneration.Manual) id: UUID)
+
+        implicit val t: Transformer[String, UUID] = str => UUID.fromString(str)
+
+        intercept[SdlIdNotProvidedException] { EntityDTO(None).transformInto[Entity] }
+        intercept[SdlIdNotProvidedException] { EntityDTO(None).into[Entity].transform }
+      }
+
+      "fail compilation if no UUID transformer in scope" - {
```

Comment:
I don't understand this test now

Reply:
Automapper has this transformer defined, so the enduser doesn't need to define one. 
However here I want to be sure my impl didn't break the transformer resolution rules

Reply:
What exactly are you testing? Don't we allow String=>UUID?

Reply:
I'm testing that my implementation does indeed look in the implicit scope for the String=>UUID transformer, not performs this transformation inside the macro somehow.

Reply:
👍